var/spack/repos/builtin/packages/openmpi/package.py

```diff
@@ -66,10 +66,11 @@ class Openmpi(AutotoolsPackage):
     version('develop', branch='master')
 
     # Current
-    version('4.0.0', sha256='2f0b8a36cfeb7354b45dda3c5425ef8393c9b04115570b615213faaa3f97366b')  # libmpi.so.40.20.0
+    version('4.0.1', sha256='cce7b6d20522849301727f81282201d609553103ac0b09162cf28d102efb9709')  # libmpi.so.40.20.1
 
     # Still supported
-    version('3.1.3', preferred=True, sha256='8be04307c00f51401d3fb9d837321781ea7c79f2a5a4a2e5d4eaedc874087ab6')  # libmpi.so.40.10.3
```

Comment:
noticed that `netlib-scalapack-2.0.2` does not like the new version, which is not a surprise as it was released in 2012. But the problem is that it's a default provider for ScaLAPACK. So perhaps we should keep `3.1.3` as `preferred`.

Reply:
or maybe turn on the deprecated funcitons by default?

Reply:
here's a relevant discussion https://github.com/open-mpi/ompi/issues/6114#issuecomment-446279495

Reply:
something like:
```
diff --git a/var/spack/repos/builtin/packages/openmpi/package.py b/var/spack/repos/builtin/packages/openmpi/package.py
index 48b255b0b..480062dc1 100644
--- a/var/spack/repos/builtin/packages/openmpi/package.py
+++ b/var/spack/repos/builtin/packages/openmpi/package.py
@@ -388,6 +388,12 @@ class Openmpi(AutotoolsPackage):
         if spec.satisfies('@3.0.0:', strict=True):
             config_args.append('--with-zlib={0}'.format(spec['zlib'].prefix))

+        # some scientific packages ignore deprecated/remove symbols. Re-enable
+        # them for now, for discussion see
+        # https://github.com/open-mpi/ompi/issues/6114#issuecomment-446279495
+        if spec.satisfies('@4.0.1:'):
+            config_args.append('--enable-mpi1-compatibility')
+
         # Fabrics
         config_args.extend(self.with_or_without('fabrics'))
         # Schedulers
```
?

Reply:
given 

> Note that the netlib scalapack folks may not have gotten the memo, but starting with the next major release of Open MPI, these deprecated MPI symbols will not be present

please limit the `if` to `spec.satisfies('@4.0')`, that should do exactly what we need.

Reply:
@davydden: Preferred version is back to 3.1.3. Thanks for the help.